packages/buidler-core/src/builtin-tasks/compile.ts

```diff
@@ -281,38 +293,48 @@ function invalidateCacheMissingArtifacts(
 }
 
 function buildCompilationGroupsFailureMessage(
-  compilationGroupsFailure: CompilationGroupsFailure
+  compilationGroupsFailures: CompilationGroupsFailure[]
```

Comment:
This changed because now you have a `CompilationGroupsFailure` for each connected component of the dependency graph.